### PR TITLE
Remove opengl, vulkan, va use for dummy compositor

### PR DESCRIPTION
**Pull request by csravelar**
Adjust makefiles in order to build without certain functionality that
we won't need enabled in the splash screen with dummy compositor flag set.

Jira: None.
Test: Build passes when using dummy compositor flag
Signed-off-by: Richard Avelar richard.avelar@intel.com

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -42,6 +42,17 @@ AC_PROG_LIBTOOL
 
 AC_SUBST(ASSEMBLER_WARN_CFLAGS)
 
+# For splash screen dummy compositor
+AC_ARG_ENABLE(dummy-compositor,
+  AS_HELP_STRING([--enable-dummy-compositor],
```
Comment (strassek) [inline configure.ac]: The issue I see here is that each backend is mutually exclusive. What would happen if I passed both "--enable-vulkan" and "--enable-dummy-compositor"? I think a better approach would be to have a '--with-compositor=' option that could take one of "gl,vulkan,dummy", at least until we get runtime selection working.

File: common/compositor/compositordefs.h [hunk 2]
```diff
@@ -42,7 +42,15 @@ static float TransformMatrices[] = {
 };
 // clang-format on
 
-#ifdef USE_GL
+#ifdef USE_DC
+typedef unsigned GpuResourceHandle;
+typedef struct gl_import {
```
[the remark this answers is not in the document]
Comment (kalyankondapally) [reply common/compositor/compositordefs.h]: hmm ya makes sense.

File: configure.ac (hunk 1 above)
Comment (kalyankondapally) [reply configure.ac]: hmm good point. @csravelar Can you do that change ?

Comment (kalyankondapally) [reply configure.ac]: Or other way around would be to have checks that only one backend is requested and throw out an error if thats not the case.

File: common/compositor/factory.cpp [hunk 3]
```diff
@@ -17,7 +17,9 @@
 #include "factory.h"
 #include "platformdefines.h"
 
-#ifdef USE_GL
+#ifdef USE_DC
+#include "nativesurface.h"
```
Comment (kalyankondapally) [inline common/compositor/factory.cpp]: CreateMediaRenderer still returns VARenderer. 3D Renderer returns NULL.Did we miss changes in factory class ?

**Pull request closed without merging**